_posts/2017-11-08-HITCON-CTF-2017.md

```diff
@@ -66,3 +66,92 @@ The flag is found in stream 42, which can be filtered for with the expression `t
 > Have you tried pwntools-ruby?  
 > nc 54.65.72.116 31337  
 > [start-7a3715a6bd61783850f9c109dbc57571.zip](/assets/hitcon2017/start-7a3715a6bd61783850f9c109dbc57571.zip)
+
+Once connected, the server serving the `server.rb` script makes it clear that the start binary is hosted locally and that a ruby script is to be provided for it to run.
+
+![server.rb](/assets/hitcon2017/serverrb.png)
+
+Quickly looking over the `server.rb` script it seems to import the pwntools library - knowing this let's start reversing and creating an exploit for the `start` binary using pwntools, to hopefully later on send the script to the server.
+
+A stack canary is put on the stack preventing us from conventional buffer overflows or similar things that tamper with the stack. 
+
+![stack canary](/assets/hitcon2017/canary.png)
+
+Luckily there is a way to leak the canary abusing the puts() call. Reading in 24 characters and a newline char `\n`, puts() will print the canary except the last char which is overwritten by the `\n`. Luckily, the last char of the canary is a null-byte in this case so it won't matter.
+
+
+Once the canary is obtained it is a simple matter of creating the exploit. I am sure there are multiple paths to success once we control the return address, I chose to go with `mprotect()` which is a syscall allowing us to overwrite permissions on the stack, making it executable (the binary has DEP/NX enabled by default).
+
+Using an already present function in this static binary `_dl_make_stack_executable` which is essentially a wrapper around the mprotect() syscall the stack is made executable, a payload is generated with `msfvenom` and the exploit is mounted.
```

Comment:
This sentence is hard to understand. Can you break this up into two or more sentences? Also, backticks around function names please.